examples/grafana.build.sh

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env bash
+
+#
+# A rkt container pod for running a Statsd/Graphite/Grafana metrics stack.
+#
+# Start the containers with new data storage directory:
+#
+#   mkdir data
+#   sudo rkt run \
+#     --port=carbon:2003 \
+#     --port=statsd:8125 \
+#     --port=graphite-api:8888 \
+#     --port=grafana-www:3000 \
+#     --volume data,kind=host,source=$(pwd)/data \
+#     woofwoofinc.dog/grafana-carbon --mount volume=data,target=/opt/graphite/storage/whisper \
+#     woofwoofinc.dog/grafana-statsd \
+#     woofwoofinc.dog/grafana-graphite-api --mount volume=data,target=/srv/graphite/whisper \
+#     woofwoofinc.dog/grafana-www --mount volume=data,target=/usr/share/grafana/data
+#
+# Create some test metric data from the macOS host:
+#
+#    brew install watch
+#    watch -n5 'echo "1d6:$(jot -r 1 1 6)|g" | nc -u -w0 rktmachine.local 8125'
+#
+# This metric can be inspected in a browser on the macOS host:
+#
+#    open 'http://rktmachine.local:8888/render?target=stats.gauges.1d6&from=-5min'
+#
+# Or by curl for the data in CSV format:
+#
+#    curl 'http://rktmachine.local:8888/render?target=stats.gauges.1d6&from=-5min&format=csv'
+#
+# The Grafana instance runs at http://rktmachine.local:3000 with default
+# `admin` user with `admin` password.
+#
+# To configure the datasource in Grafana, select a Proxy source with URL
+# http://0.0.0.0:8888. Using a proxy source means the graphite-api service does
+# not have to be exposed on port 8888. Normally this is preferred because it is
+# easier to secure the Grafana endpoint for authorized use.
+#
+# We expose the graphite-api port in the rkt run command here because it is
+# convenient for data export. The datasource would still work if
+# `--port=graphite-api:8888` was omitted.
+#
+# Similarly the Carbon `--port=carbon:2003` could be omitted but again it may
+# be useful during development.
+#
+# Useful links for writing Carbon aggregation rules.
+# - https://github.com/etsy/statsd/blob/master/docs/graphite.md
+# - http://dieter.plaetinck.be/post/25-graphite-grafana-statsd-gotchas/
+#
+
+set -x
+
+GRAFANA_VERSION=4.2.0
+NODE_VERSION=6.10.2
+STATSD_VERSION=0.8.0
+
+
+################################################################################
+# Setup
+################################################################################
+
+mkdir -p grafana
+cd grafana
```

Comment:
Something I've recently learned: `pushd` and `popd` are kinda cool. If you replace `cd grafana` and the like below with `pushd grafana` and `cd ..` with `popd` it's less likely you accidentally `cd` to the wrong place in between and `rm` something you didn't mean to in the cleanup step.